library/src/scala/quoted/Expr.scala

```diff
@@ -190,6 +190,11 @@ package internal {
      *  May contain references to code defined outside this TastyTreeExpr instance.
      */
     final class TastyTreeExpr[Tree](val tree: Tree, val scopeId: Int) extends Expr[Any] {
+      override def equals(that: Any): Boolean = that match {
+        case that: TastyTreeExpr[_] => tree == that.tree && scopeId == that.scopeId
+        case _ => false
+      }
```

Comment:
Maybe add a comment about what we can expect if two Expr are equal. For example, why referential equality is not good enough?

Reply:
Added a comment